code/target/src/androidTest/java/com/adobe/marketing/mobile/target/TargetFunctionalTests.kt

```diff
@@ -147,68 +164,380 @@ class TargetFunctionalTests {
+            dataStore?.removeAll()
         }
     }
 
     @Before
     fun setup() {
+        setupNetwork()
         MobileCore.setApplication(ApplicationProvider.getApplicationContext())
         MobileCore.setLogLevel(LoggingMode.VERBOSE)
-        val countDownLatch = CountDownLatch(1)
+
+        // Register required extensions
+        val waitForRegistration = CountDownLatch(1)
         MobileCore.registerExtensions(
             listOf(
                 Target.EXTENSION,
+                Identity.EXTENSION,
                 MonitorExtension::class.java
             )
         ) {
-            countDownLatch.countDown()
+            waitForRegistration.countDown()
         }
-        Assert.assertTrue(countDownLatch.await(1000, TimeUnit.MILLISECONDS))
+        assertTrue(waitForRegistration.await(1000, TimeUnit.MILLISECONDS))
 
+        // Configuration
         val configurationLatch = CountDownLatch(1)
         configurationAwareness { configurationLatch.countDown() }
         MobileCore.updateConfiguration(getConfigurationData())
-        Assert.assertTrue(configurationLatch.await(1000, TimeUnit.MILLISECONDS))
+        assertTrue(configurationLatch.await(1000, TimeUnit.MILLISECONDS))
+
+        // set latches
+        resetLatches()
+
+        // setup network capturer
+        networkMonitor = { request ->
+            networkRequestBody = String(request.body, Charsets.UTF_8)
+            waitForNetworkCall?.countDown()
+        }
     }
 
     @After
     fun tearDown() {
-        targetResponse = null
-        Target.resetExperience()
         Target.clearPrefetchCache()
+        Target.resetExperience()
+        resetNetworkVariables()
+        resetCallbackResponses()
+        clearDatastore()
+        TargetTestHelper.cleanCacheDir()
+        SDKHelper.resetSDK()
+        SDKHelper.resetTargetListener()
     }
 
     //1
     @Test
     fun testExtensionVersion() {
-        Assert.assertEquals(TargetTestConstants.EXTENSION_VERSION, Target.extensionVersion())
+        assertEquals(TargetTestConstants.EXTENSION_VERSION, Target.extensionVersion())
     }
 
     // ---------------------------------------------------//
     // Happy Tests
     // ---------------------------------------------------//
+    // Test Case No : 1
+    @Test
+    @Throws(Exception::class)
+    fun test_Functional_Happy_Target_targetRetrieveLocationContent_VerifyWithDefaultParameters() {
+        // test
+        val targetParameters = TargetParameters.Builder()
+            .parameters(mboxParameters)
+            .profileParameters(profileParameters)
+            .build()
+        val targetRequest = TargetRequest(mboxName, targetParameters , defaultContent) { data ->
+            retrievedLocationResponse = data
+            waitForCallback?.countDown()
+        }
+        Target.retrieveLocationContent(listOf(targetRequest), targetParameters)
+        waitForCallback?.await(50, TimeUnit.SECONDS)
```

Comment:
nit: should this be in seconds? or could the value be lowered?

Reply:
Ah blunder.. I used this while debugging and forgot to revert back